File: examples/read.jl
using DataFrames, CSV

pa = "nhanes"

fn = ["DEMO_J", "BMX_J", "BPX_J"]
dl = []
for f in fn
    dx = open(joinpath(pa, "$(f).csv.gz")) do io
        CSV.read(io, DataFrame)
    end
    push!(dl, dx)
end

df = dl[1]
df = leftjoin(df, dl[2], on = :SEQN)
df = leftjoin(df, dl[3], on = :SEQN)

df[!, :RIAGENDR] = replace(df[:, :RIAGENDR], 1 => "M", 2 => "F")
df = filter(r -> r.RIDAGEYR >= 18, df)

File: examples/read.R
library(dplyr)
library(readr)

pa = "nhanes"
fn = c("DEMO_J.csv.gz", "BMX_J.csv.gz", "BPX_J.csv.gz")

df = NULL
for (f in fn) {
    dx = read_csv(file.path(pa, f))
    if (is.null(df)) {
        df = dx
    } else {
        df = left_join(df, dx, "SEQN")
    }
}

df = df %>% mutate(RIAGENDR=recode(df$RIAGENDR, `1`="M", `2`="F"))

df = df %>% filter(RIDAGEYR >= 18)
